tket/src/Mapping/MappingFrontier.cpp

```diff
@@ -257,160 +260,196 @@ void MappingFrontier::advance_frontier_boundary(
+         *
+         * If the "in edge" is Boolean then the vertex may edit, replace or
+         * remove a boolean bundle. The "in bundle" should never be empty. We
+         * construct a new candidate "out bundle" by combining the vertex "out
+         * bundle" (for this port) with any other boolean edges the boolean
+         * boundary has for this Bit that are not connected to this vertex. If
+         * the candidate "out bundle" is empty, we erase the Bit from the
+         * boundary. Else, we replace the "out edges" stored in the boundary for
+         * this Bit.
+         */
+
+        for (port_t port = 0; port < n_in_bundles; port++) {
+          std::pair<UnitID, EdgeType> linear_uid = in_uids[port];
+          EdgeVec in_bundle = in_bundles[port];
+          EdgeVec out_bundle = out_bundles[port];
+          switch (linear_uid.second) {
+            // Linear & can't spawn Boolean bundle: pass
+            case EdgeType::Quantum:
+              break;
+            // Linear but can spawn Boolean bundle
+            // If Boolean edges spawned, add to boolean boundary
+            case EdgeType::Classical: {
+              EdgeVec out_bundle = out_bundles[port];
+              TKET_ASSERT(in_bundle.empty());
+              if (!out_bundle.empty()) {
+                this->boolean_boundary->insert(
+                    {Bit(linear_uid.first), out_bundle});
+              }
+              break;
+            }
+            // Can spawn, erase or edit boolean bundles
+            case EdgeType::Boolean: {
+              TKET_ASSERT(!in_bundle.empty());
+              Bit bit = Bit(linear_uid.first);
+              auto boolean_it = this->boolean_boundary->get<TagKey>().find(bit);
+              TKET_ASSERT(
+                  boolean_it != this->boolean_boundary->get<TagKey>().end());
+              // update "out bundle" with other boolean edges attached to other
+              // vertices
+              for (const Edge& edge : boolean_it->second) {
+                if (std::find(in_bundle.begin(), in_bundle.end(), edge) ==
+                    in_bundle.end()) {
+                  out_bundle.push_back(edge);
+                }
+              }
+              if (out_bundle.empty()) {
+                // => all edges in boolean bundle in boolean boundary attached
+                // to this vertex
+                // => vertex has no edges in output boolean bundle for thos port
+                // => can erase Bit from boolean boundary as its not longer used
+                this->boolean_boundary->erase(boolean_it);
+              } else {
+                // => either Vertex has edges in output boolean bundle
+                // => or there are other edges in boolean bundle held in boolean
+                // boundary that are not attached to this vertex
+                // => update boolean boundary
+                this->boolean_boundary->replace(boolean_it, {bit, out_bundle});
+              }
+              break;
```

Comment:
Add `default` case with `TKET_ASSERT`.

Reply:
Oops, I thought I did these locally but it turns out my local file changes weren't being tracked by github for some reason. All done now.